test/test_framework.py

```diff
@@ -718,6 +729,12 @@ def test_helper_properties(self):
         self.assertEqual(my_obj.meta, framework.meta)
         self.assertEqual(my_obj.charm_dir, framework.charm_dir)
 
+    def test_ban_concurrent_frameworks(self):
+        _ = self.create_framework()  # noqa: F841
```

Comment:
Isn't this unsafe because you don't know what is referencing the framework object, so Python might GC the reference?
You can put a f.close() at the end, so that flake doesn't feel like you have an unused variable.

Reply:
AFAIU it will be kept because we have a cleanup step that auto-closes `Framework` instances created by `create_framework` https://github.com/canonical/operator/blob/master/test/test_framework.py#L728-L730.

I don't see any harm in adding an explicit `close` in the test case for clarity though.

Reply:
it's not go: `_` is an actual variable, so the framework object is referenced by it, so it won't get GC'ed.

OTOH doing a `.close` feels cleaner.

Reply:
Addressed in https://github.com/canonical/operator/pull/141/commits/8ec7e1fb04dc416f108fb20715a9953c132781e9